src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```
@@ -0,0 +1,43 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Dao;
+import mate.academy.lib.Inject;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Dao
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email);
+        if (user.getUsername() == null) {
```

You are checking for `user.getUsername() == null` to determine if the user exists. The requirements specify that you should use `Optional<User>` and throw `AuthenticationException` for both user not found and password mismatch, using a single condition. This logic is not compliant.